subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1368,8 +1368,7 @@ int bt_bap_scan_delegator_set_bis_sync_state(
 		internal_state->index, src_id);
 
 	if (internal_state->state.encrypt_state == BT_BAP_BIG_ENC_STATE_BAD_CODE) {
-		(void)memcpy(internal_state->state.bad_code,
-			     internal_state->broadcast_code,
+		(void)memset(internal_state->state.bad_code, 0xFF,
```

Comment:
Your changes look good in terms of fixing the issue, so based on that I'm OK with approving this. 

One thing that I see as an optimization is that since the bad_code field can only ever be all-0xFF, then we could actually remove the `bad_code` field from the `bt_bap_scan_delegator_recv_state` and then just populate the `read_buf` with 16 * `0xFF` when it is being dumped, read or notified. 

That way we can save 16 octets of RAM per receive state

If that makes sense to you, and you want to update the PR to reflect that, that'd be great, but otherwise we can also keep it simple for now and just focus on fixing the issue.

Reply:
If possible to add to this PR, it would be great - otherwise, maybe a quick followup?